src/hyp_map.rs

```diff
@@ -19,21 +19,21 @@ use riscv_pages::{
 use riscv_regs::{satp, sstatus, LocalRegisterCopy, ReadWriteable, SatpHelpers, CSR};
 use sync::Once;
 
-// The copy to/from guest memory routines defined in umode_mem.S.
+// The copy to/from guest memory routines defined in extable.S.
 extern "C" {
     fn _copy_to_user(dest_addr: u64, src: *const u8, len: usize) -> usize;
     fn _copy_from_user(dest: *mut u8, src_addr: u64, len: usize) -> usize;
 }
 
-// Maximum number of regions unique to every pagetable (private).
-const MAX_PRIVATE_REGIONS: usize = 32;
-// Maximum number of regions shared across all pagetables.
-const MAX_SHARED_REGIONS: usize = 32;
+// Maximum number of U-mode ELF regions.
+const MAX_UMODE_ELF_REGIONS: usize = 32;
+// Maximum number of hardware map regions.
+const MAX_HW_MAP_REGIONS: usize = 32;
```

Comment:
these regions are that are identity mapped, right? might be more obvious if this were called the "linear map" or "identity map"

Reply:
Was future-proofing for #180, and in general other possible identity mapped might require a different structure, and this change was exactly to create a region type for each source.

Reply:
Haven't changed this. Let me know if you have strong opinions about this.